core/src/main/java/org/elasticsearch/rest/AbstractRestChannel.java

```diff
@@ -58,28 +57,26 @@ protected AbstractRestChannel(RestRequest request, boolean detailedErrorsEnabled
 
     @Override
     public XContentBuilder newBuilder() throws IOException {
-        return newBuilder(request.hasContent() ? request.content() : null, true);
+        return newBuilder(request.getXContentType(), true);
     }
 
     @Override
     public XContentBuilder newErrorBuilder() throws IOException {
         // Disable filtering when building error responses
-        return newBuilder(request.hasContent() ? request.content() : null, false);
+        return newBuilder(request.getXContentType(), false);
     }
 
     @Override
-    public XContentBuilder newBuilder(@Nullable BytesReference autoDetectSource, boolean useFiltering) throws IOException {
+    public XContentBuilder newBuilder(@Nullable XContentType xContentType, boolean useFiltering) throws IOException {
         XContentType contentType = XContentType.fromMediaTypeOrFormat(format);
```

Comment:
Correct, since format could be a value from the `Accept` header or the query parameter.